src/activity_item_handling.cpp

```diff
@@ -1133,14 +1187,81 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
     // see activity_handlers.h cant_do_activity_reason enums
     zone_manager &mgr = zone_manager::get_manager();
     std::vector<zone_data> zones;
-    // check if another NPC or player is doing anything where we want to do something.
-    if( g->m.getlocal( g->u.activity.placement ) == src_loc ) {
-        return activity_reason_info::fail( ALREADY_WORKING );
-    }
-    for( const npc &guy : g->all_npcs() ) {
-        if( g->m.getlocal( guy.activity.placement ) == src_loc ) {
+    if( act == activity_id( "ACT_VEHICLE_DECONSTRUCTION" ) ) {
+        if( g->m.getlocal( g->u.activity.placement ) == src_loc ) {
             return activity_reason_info::fail( ALREADY_WORKING );
         }
+        for( const npc &guy : g->all_npcs() ) {
+            if( g->m.getlocal( guy.activity.placement ) == src_loc || guy.pos() == src_loc ) {
+                return activity_reason_info::fail( ALREADY_WORKING );
+            }
+        }
+        vehicle *veh = veh_pointer_or_null( g->m.veh_at( src_loc ) );
+        if( !veh ) {
+            return activity_reason_info::fail( NO_ZONE );
+        } else {
+            // find out if there is a vehicle part here we can remove.
+            std::vector<vehicle_part *> parts = veh->get_parts_at( src_loc, "", part_status_flag::any );
+            for( vehicle_part *part_elem : parts ) {
+                const vpart_info &vpinfo = part_elem->info();
+                int vpindex = veh->index_of_part( part_elem, true );
+                // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
+                if( vpindex == -1 || !veh->can_unmount( vpindex ) ) {
+                    continue;
+                }
+                for( const npc &guy : g->all_npcs() ) {
+                    if( guy.disp_name() != p.disp_name() && guy.activity_vehicle_part_index != -1 &&
+                        guy.activity_vehicle_part_index == vpindex ) {
+                        continue;
+                    }
+                }
+                // if the vehicle is moving or player is controlling it.
+                if( abs( veh->velocity ) > 100 || veh->engine_on ) {
+                    continue;
+                }
+                // dont have skill to remove it
+                std::map<skill_id, int> removal_skills = vpinfo.removal_skills;
+                for( const auto &e : removal_skills ) {
+                    bool hasSkill = p.get_skill_level( e.first ) >= e.second;
+                    if( !hasSkill ) {
+                        continue;
+                    }
+                }
+                item base( vpinfo.item );
+                if( base.is_wheel() ) {
+                    // no wheel removal yet
```

Comment:
Why. This is probably causing #39153 and there is no explanation for that arbitrary limitation.

Reply:
The PR is authored by a 'ghost', so you're unlikely to get any answers... :(